File: app/src/main/java/com/vmenon/mpo/util/MediaHelper.java
package com.vmenon.mpo.util;

import com.vmenon.mpo.api.Episode;

public class MediaHelper {

    public static class MediaType {
        int mediaType;
        long id;

        public MediaType(int mediaType, long id) {
            this.mediaType = mediaType;
            this.id = id;
        }

        public int getMediaType() {
            return mediaType;
        }

        public long getId() {
            return id;
        }
    }

    public static final int MEDIA_TYPE_EPISODE = 0;

    private static final String EPISODE_MEDIA_PREFIX = "episode";

    public static String createMediaId(Episode episode) {
        return EPISODE_MEDIA_PREFIX + ":" + episode.id;
    }

    public static MediaType getMediaTypeFromMediaId(String mediaId) {
        String[] mediaIdParts = mediaId.split(":");
        if (EPISODE_MEDIA_PREFIX.equals(mediaIdParts[0])) {
            long episodeId = Long.parseLong(mediaIdParts[1]);
            return new MediaType(MEDIA_TYPE_EPISODE, episodeId);
        }

        return null;
    }
}

File: app/src/main/java/com/vmenon/mpo/util/MediaHelper.kt
package com.vmenon.mpo.util

import com.vmenon.mpo.api.Episode

object MediaHelper {

    const val MEDIA_TYPE_EPISODE = 0

    private const val EPISODE_MEDIA_PREFIX = "episode"

    class MediaType(mediaType: Int, id: Long) {
        var mediaType: Int = 0
            internal set
        var id: Long = 0
            internal set

        init {
            this.mediaType = mediaType
            this.id = id
        }
    }

    fun createMediaId(episode: Episode): String {
        return EPISODE_MEDIA_PREFIX + ":" + episode.id
    }

    fun getMediaTypeFromMediaId(mediaId: String): MediaType? {
        val mediaIdParts = mediaId.split(":".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        if (EPISODE_MEDIA_PREFIX == mediaIdParts[0]) {
            val episodeId = java.lang.Long.parseLong(mediaIdParts[1])
            return MediaType(MEDIA_TYPE_EPISODE, episodeId)
        }

        return null
    }
}
